backend/app/app/tools/monitoring_callbacks.py

```diff
@@ -3,27 +3,38 @@
 def location_callback(ue, callbackurl, subscription):
     url = callbackurl
 
-    payload = json.dumps({
-    "externalId" : ue.get("external_identifier"),
-    "ipv4Addr" : ue.get("ip_address_v4"),
-    "subscription" : subscription,
-    "monitoringType": "LOCATION_REPORTING",
-    "locationInfo": {
-        "cellId": ue.get("cell_id_hex"),
-        "enodeBId": ue.get("gnb_id_hex")
-    }
-    })
-    headers = {
-    'accept': 'application/json',
-    'Content-Type': 'application/json'
-    }
+    payload = json.dumps(
+        {
+            "subscription": subscription,
+            "monitoringEventReports": [
+                {
+                    "externalId": ue.get("external_identifier"),
+                    "monitoringType": "LOCATION_REPORTING",
+                    "locationInfo": {
+                        "cellId": ue.get("cell_id_hex"),
+                        "enodeBId": ue.get("gnb_id_hex"),
+                        "geographicArea": {
+                            "shape": "POINT",
+                            "point": {
+                                "lat": ue.get("latitude"),
+                                "lon": ue.get("longitude"),
+                            },
+                        },
+                    },
+                }
+            ],
+        }
+    )
```

Comment:
a more consistent approach would be to rely on a class (for instance, a pydantic schema) to offer this payload.
you may rely on the `MonitoringEventReport` schema.